generators/upgrade/index.js

```diff
@@ -224,7 +224,7 @@ module.exports = UpgradeGenerator.extend({
         updateJhipster: function() {
             this.log(chalk.yellow('Updating ' + GENERATOR_JHIPSTER + '. This might take some time...'));
             var done = this.async();
-            shelljs.exec('npm install ' + GENERATOR_JHIPSTER, {silent:true}, function (code, msg, err) {
+            shelljs.exec('npm install ' + GENERATOR_JHIPSTER + '@' + this.latestVersion, {silent:true}, function (code, msg, err) {
```

Comment:
its easy just use `yarn add` for yarn and skip any gulp tasks if app is using angular 2 as the yarn install step automatically will trigger webpack build during app regeneration